From 519f2bcb0b0712c6839d7a3b7c06d0a0cac0e93d Mon Sep 17 00:00:00 2001
From: Udit Kumar <udit.kumar@nxp.com>
Date: Wed, 12 Jun 2019 10:42:02 +0530
Subject: [PATCH 235/235] ahci_qoriq: bug fix for ecc_addr

Original driver expect a register node with name "sata-ecc"
this node is of 64 bit wide.
In ACPI such nodes can be provided with QWordMemory, but
QWordMemory can not hold DescriptorName more than 4 characters.

Therefore this patch changes platform property retrival based
upon index instead of named.

Signed-off-by: Udit Kumar <udit.kumar@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/ata/ahci_qoriq.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/ata/ahci_qoriq.c b/drivers/ata/ahci_qoriq.c
index a2483eb17c95..1ed3c2c02181 100644
--- a/drivers/ata/ahci_qoriq.c
+++ b/drivers/ata/ahci_qoriq.c
@@ -437,9 +437,7 @@ static int ahci_qoriq_probe(struct platform_device *pdev)
 		qoriq_priv->type = (enum ahci_qoriq_type)acpi_id->driver_data;
 
 	if (unlikely(!ecc_initialized)) {
-		res = platform_get_resource_byname(pdev,
-						   IORESOURCE_MEM,
-						   "sata-ecc");
+		res = platform_get_resource(pdev, IORESOURCE_MEM, 1);
 		if (res) {
 			qoriq_priv->ecc_addr =
 				devm_ioremap_resource(dev, res);
-- 
2.17.1

